src/components/core/base/VisualizationLoadingHOC.tsx

```diff
@@ -20,17 +35,22 @@ import { ErrorComponent, IErrorProps } from '../../simple/ErrorComponent';
 import { RuntimeError } from '../../../errors/RuntimeError';
 import { IPushData } from '../../../interfaces/PushData';
 import { IChartConfig } from '../../../interfaces/Config';
+import { setTelemetryHeaders } from '../../../helpers/utils';
+
+const escapeFileName = (str: string) => str && str.replace(/[\/\?<>\\:\*\|":]/g, '');
 
 export type IExecutionDataPromise = Promise<Execution.IExecutionResponses>;
 
 export interface ICommonVisualizationProps extends IEvents {
+    projectId?: string;
     locale?: string;
     drillableItems?: Array<IDrillableItem | IHeaderPredicate>;
     afterRender?: () => void;
     pushData?: (data: IPushData) => void;
     ErrorComponent?: React.ComponentType<IErrorProps>;
     LoadingComponent?: React.ComponentType<ILoadingProps>;
     config?: IChartConfig;
+    onExportReady?: OnExportReady;
```

Comment:
I would rather put it into IEvents interface where all other "callback" props are defined